policy/modules/services/container.te

```diff
@@ -0,0 +1,880 @@
+	fs_read_nfs_symlinks(container_engine_domain)
+	fs_mount_nfs(container_engine_domain)
+	fs_unmount_nfs(container_engine_domain)
+	fs_exec_nfs_files(container_engine_domain)
+	kernel_rw_fs_sysctls(container_engine_domain)
+')
+
+tunable_policy(`container_use_samba',`
+	fs_manage_cifs_dirs(container_engine_domain)
+	fs_manage_cifs_files(container_engine_domain)
+	fs_manage_cifs_named_sockets(container_engine_domain)
+	fs_read_cifs_symlinks(container_engine_domain)
+	fs_exec_cifs_files(container_engine_domain)
+')
+
+optional_policy(`
+	# to verify container image signatures
+	gpg_exec(container_engine_domain)
+	gpg_dontaudit_exec_agent(container_engine_domain)
+	gpg_dontaudit_search_user_secrets(container_engine_domain)
+')
+
+optional_policy(`
+	iptables_domtrans(container_engine_domain)
+')
+
+########################################
+#
+# Common system container engine local policy
+#
+
+# system container engines can interact with any container
+ps_process_pattern(container_engine_system_domain, container_domain)
+allow container_engine_system_domain container_domain:process { sigkill signal signull transition };
+allow container_engine_system_domain container_domain:key { create search setattr view };
+
+create_dirs_pattern(container_engine_system_domain, container_config_t, container_config_t)
+files_etc_filetrans(container_engine_system_domain, container_config_t, dir, "cni")
+files_etc_filetrans(container_engine_system_domain, container_config_t, dir, "containers")
+files_etc_filetrans(container_engine_system_domain, container_config_t, dir, "docker")
+
+manage_dirs_pattern(container_engine_system_domain, container_log_t, container_log_t)
+manage_files_pattern(container_engine_system_domain, container_log_t, container_log_t)
+logging_log_filetrans(container_engine_system_domain, container_log_t, { dir file })
+
+allow container_engine_system_domain container_var_lib_t:dir { manage_dir_perms relabel_dir_perms watch };
+allow container_engine_system_domain container_var_lib_t:file { manage_file_perms relabel_file_perms exec_file_perms };
+allow container_engine_system_domain container_var_lib_t:blk_file { manage_blk_file_perms relabel_blk_file_perms };
+allow container_engine_system_domain container_var_lib_t:chr_file { manage_chr_file_perms relabel_chr_file_perms };
+allow container_engine_system_domain container_var_lib_t:fifo_file { manage_fifo_file_perms relabel_fifo_file_perms };
+allow container_engine_system_domain container_var_lib_t:lnk_file { manage_lnk_file_perms relabel_lnk_file_perms };
+allow container_engine_system_domain container_var_lib_t:sock_file { manage_sock_file_perms relabel_sock_file_perms };
+files_var_lib_filetrans(container_engine_system_domain, container_var_lib_t, dir, "containers")
+files_var_lib_filetrans(container_engine_system_domain, container_var_lib_t, dir, "cni")
+files_var_lib_filetrans(container_engine_system_domain, container_var_lib_t, dir, "docker")
```

Comment:
I don't see type transitions that conflicts; did I miss something? If there isn't then these don't need names.

Reply:
Named transitions are not required here, so I will fix this. I originally wrote it this way for clarity but I agree it is not necessary.